src/devices/hid-keyboard.c

```diff
@@ -0,0 +1,188 @@
+/*
+i2c-keyboard.c - HID Keyboard
+Copyright (C) 2022  X512 <github.com/X547>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "i2c-hid.h"
+#include "hid_api.h"
+#include "spinlock.h"
+#include "utils.h"
+#include "bit_ops.h"
+
+#define MAX_PRESSED_KEYS 6
+
+static const uint8_t keyboard_hid_report_descriptor[] = {
+    0x05, 0x01,     /* Usage Page (Generic Desktop) */
+    0x09, 0x06,     /* Usage (Keyboard) */
+    0xa1, 0x01,     /* Collection (Application) */
+    0x75, 0x01,     /*   Report Size (1) */
+    0x95, 0x08,     /*   Report Count (8) */
+    0x05, 0x07,     /*   Usage Page (Key Codes) */
+    0x19, 0xe0,     /*   Usage Minimum (224) */
+    0x29, 0xe7,     /*   Usage Maximum (231) */
+    0x15, 0x00,     /*   Logical Minimum (0) */
+    0x25, 0x01,     /*   Logical Maximum (1) */
+    0x81, 0x02,     /*   Input (Data, Variable, Absolute) */
+    0x95, 0x01,     /*   Report Count (1) */
+    0x75, 0x08,     /*   Report Size (8) */
+    0x81, 0x01,     /*   Input (Constant) */
+    0x95, 0x05,     /*   Report Count (5) */
+    0x75, 0x01,     /*   Report Size (1) */
+    0x05, 0x08,     /*   Usage Page (LEDs) */
+    0x19, 0x01,     /*   Usage Minimum (1) */
+    0x29, 0x05,     /*   Usage Maximum (5) */
+    0x91, 0x02,     /*   Output (Data, Variable, Absolute) */
+    0x95, 0x01,     /*   Report Count (1) */
+    0x75, 0x03,     /*   Report Size (3) */
+    0x91, 0x01,     /*   Output (Constant) */
+    0x95, 0x06,     /*   Report Count (6) */
+    0x75, 0x08,     /*   Report Size (8) */
+    0x15, 0x00,     /*   Logical Minimum (0) */
+    0x25, 0xff,     /*   Logical Maximum (255) */
+    0x05, 0x07,     /*   Usage Page (Key Codes) */
+    0x19, 0x00,     /*   Usage Minimum (0) */
+    0x29, 0xff,     /*   Usage Maximum (255) */
+    0x81, 0x00,     /*   Input (Data, Array) */
+    0xc0,       /* End Collection */
+};
+
+struct hid_keyboard {
+    hid_dev_t hid_dev;
+    
+    uint32_t pad[1024];
```

Comment:
What's this for? I see no more references to this, alignment sorta or debugging remnants?

Reply:
Debug leftover. Removed.